zrml/swaps/src/tests.rs

```diff
@@ -1365,6 +1381,39 @@ fn swap_exact_amount_in_exchanges_correct_values_with_cpmm() {
     });
 }
 
+#[test]
+fn swap_exact_amount_in_fails_if_min_asset_amount_out_is_not_satisfied_with_cpmm() {
+    ExtBuilder::default().build().execute_with(|| {
+        create_initial_pool_with_funds_for_alice(ScoringRule::CPMM, true);
+        let expected_amount = 9900990100;
+        assert_noop!(
+            Swaps::swap_exact_amount_in(
+                alice_signed(),
+                0,
+                ASSET_A,
+                _1,
+                ASSET_B,
+                Some(expected_amount + 100),
+                None,
+            ),
+            crate::Error::<Runtime>::LimitOut,
+        );
+    });
+}
+
+#[test]
+fn swap_exact_amount_in_fails_if_max_price_is_not_satisfied_with_cpmm() {
+    ExtBuilder::default().build().execute_with(|| {
+        create_initial_pool_with_funds_for_alice(ScoringRule::CPMM, true);
+        // We're swapping 1:1, but due to slippage the price will exceed _1, so this should raise an
+        // error:
+        assert_noop!(
+            Swaps::swap_exact_amount_in(alice_signed(), 0, ASSET_A, _1, ASSET_B, None, Some(_1)),
```

Comment:
I don't really get this. Could you please explain this simply like I am a five year old?